internal/pkg/application/measurements/measurements.go

```diff
@@ -0,0 +1,305 @@
+	WatermeterURN   string = "urn:oma:lwm2m:ext:3424"
+)
+
+func GetMeasurementType(m iotCore.MessageAccepted) string {
+	typeOfMeasurement := m.BaseName()
+
+	for _, m := range m.Pack {
+		if strings.EqualFold(m.Name, "env") && m.StringValue != "" {
+			return typeOfMeasurement + "/" + m.StringValue
+		}
+	}
+
+	return typeOfMeasurement
+}
+
+func AirQualityObserved(ctx context.Context, msg iotCore.MessageAccepted, cbClient client.ContextBrokerClient) error {
+	properties := make([]entities.EntityDecoratorFunc, 0, 5)
+
+	properties = append(properties,
+		decorators.DateObserved(msg.Timestamp),
+	)
+
+	const (
+		SensorValue   int = 5700
+		CarbonDioxide int = 17
+	)
+
+	temp, tempOk := iotCore.Get[float64](msg, TemperatureURN, SensorValue)
+	if tempOk {
+		properties = append(properties, Temperature(temp, time.Unix(int64(msg.BaseTime()), 0)))
+	}
+
+	co2, co2Ok := iotCore.Get[float64](msg, AirQualityURN, CarbonDioxide)
+	if co2Ok {
+		properties = append(properties, CO2(co2, time.Unix(int64(msg.BaseTime()), 0)))
+	}
+
+	if !tempOk && !co2Ok {
+		return fmt.Errorf("no relevant properties were found in message from %s, ignoring", msg.Sensor)
+	}
+
+	if msg.HasLocation() {
+		properties = append(properties, decorators.Location(msg.Latitude(), msg.Longitude()))
+	}
+
+	id := fiware.AirQualityObservedIDPrefix + msg.Sensor
+
+	return cip.MergeOrCreate(ctx, cbClient, id, fiware.AirQualityObservedTypeName, properties)
+}
+
+func Device(ctx context.Context, msg iotCore.MessageAccepted, cbClient client.ContextBrokerClient) error {
+	properties := make([]entities.EntityDecoratorFunc, 0, 5)
+
+	properties = append(properties,
+		decorators.DateLastValueReported(msg.Timestamp),
+	)
+
+	const DigitalInputState int = 5500
+	if v, ok := iotCore.Get[bool](msg, PresenceURN, DigitalInputState); ok {
+		if v {
+			properties = append(properties, decorators.Status("on"))
+		} else {
+			properties = append(properties, decorators.Status("off"))
+		}
+	} else {
+		return fmt.Errorf("no relevant properties were found in message from %s, ignoring", msg.Sensor)
+	}
+
+	if msg.HasLocation() {
+		properties = append(properties, decorators.Location(msg.Latitude(), msg.Longitude()))
+	}
+
+	id := fiware.DeviceIDPrefix + msg.Sensor
+
+	return cip.MergeOrCreate(ctx, cbClient, id, fiware.DeviceTypeName, properties)
+}
+
+func GreenspaceRecord(ctx context.Context, msg iotCore.MessageAccepted, cbClient client.ContextBrokerClient) error {
+	properties := make([]entities.EntityDecoratorFunc, 0, 5)
+
+	properties = append(properties, decorators.DateObserved(msg.Timestamp))
+
+	id := fmt.Sprintf("%s%s", fiware.GreenspaceRecordIDPrefix, msg.Sensor)
+	observedBy := fmt.Sprintf("%s%s", fiware.DeviceIDPrefix, msg.Sensor)
+
+	const SensorValue int = 5700
+	if pr, ok := iotCore.Get[float64](msg, PressureURN, SensorValue); ok {
+		properties = append(properties, decorators.Number("soilMoisturePressure", pr, p.UnitCode("KPA"), p.ObservedAt(msg.Timestamp), p.ObservedBy(observedBy)))
```

Comment:
Skickas det alltså som kPa i SenML? Det kanske vi borde fixa. Ser ut som att Pa (utan k) är det som gäller där.

Reply:
nu kommer det kPa från sensorn/decodern och skickas vidare i convertern as-is. Jag kan lägga det som en bugg i iot-agent.